src/measure.rs

```diff
@@ -426,13 +426,9 @@ async fn measurement_loop(ctx: &mut MeasurementContext) -> Result<(u64, u64)> {
         }
 
         if let Some(interval_to_run) = interval_to_run.as_mut() {
-            let mut shutdown_triggered = ctx.shutdown_handler.write().await.trigger.subscribe();
             tokio::select! {
                 _ = interval_to_run.tick() => {
                 }
-                _ = shutdown_triggered.recv() => {
-                    break;
-                }
```

Comment:
Doesn't this mean a shutdown triggered while waiting for say a 30 seconds will wait the full interval before shutting down... That's not what we want, right..

Reply:
Yes you right, this line should be included again.